runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -97,6 +97,33 @@ local function tags_lsp_to_vim(diagnostic, client_id)
   return tags
 end
 
+---@param related_informations lsp.DiagnosticRelatedInformation[]
+---@param offset_encoding string
+---@return vim.DiagnosticRelatedInformation[]
+local function related_information_lsp_to_vim(related_informations, offset_encoding)
+  return vim.iter(related_informations):map(
+  --- @param related_information lsp.DiagnosticRelatedInformation
+  --- @return vim.DiagnosticRelatedInformation
+    function(related_information)
+      local fname = vim.uri_to_fname(related_information.location.uri)
+      local bufnr = vim.fn.bufadd(fname)
+      if not bufnr then
+        return
+      end
```

Comment:
It's functionally equivalent in this case but imo is slightly clearer (we explicitly return either `nil` or a `vim.DiagnosticRelatedInformation` on all code paths).

```suggestion
      local bufnr = vim.fn.bufadd(fname)
      if not bufnr then
        return nil
      end
```